### DISCUSSION TITLE
getbbox in Link SVG gives height and width 0

### DISCUSSION THREAD

**Discussion by slvamegestic** [clientIO/joint#1967] [Ideas]
```js
 var link = new joint.shapes.standard.Link();
    link.attr({
       line: {
           sourceMarker: {
               'type': 'image',
              'xlink:href': `data:image/svg+xml;base64;utf8,${btoa(this.svgFile)}`,
               'width': 150,
               'height': 150,
              'y': 50,
               'x': 675
           }
       }
    });
     this.headGraph.addCell(link)
```

I am using the above code to add a custom image as a link, but I cannot get the boundary box on this line. Using GetBBox and getClientRects() gives height and width as 0 .can you please help me on this

**Comment by kumilingus** [comment]
By default, the `SVGMarkers` are not included in the bounding box (see `getBBox()` [docs](https://developer.mozilla.org/en-US/docs/Web/API/SVGGraphicsElement/getBBox)). Unfortunately, the `markers` option does not work at the moment.

See the logs in this [example](https://jsfiddle.net/kumilingus/xofqjbc2/). The height is always `0`.

If you want the image to be included in the bounding box, use link labels or elements.

**Reply by slvamegestic** [reply]
I understand, I have seen that. Could you please explain to me how to obtain a bounding box using link labels and elements? That would be great if you made me do that since I still have not been able to find a solution to that problem

**Reply by kumilingus** [reply]
What are you trying to do? Why do you need to find the bounding box of the link which includes the marker?

**Reply by slvamegestic** [reply]
I am trying to follow the example below. There is an SVG Image that should be rendered as a link (not an image) referred to as a Bag, in which the items are to be dropped. I believe it is feasible. One of the main things to be implemented is the item should not be able to be dragged outside of the bag only inside the bag(marked a red line). Except for this SVG Line cell, I have done that using boundary coordinates.
<img width="143" alt="Screenshot 2023-01-04 at 11 20 29 AM" src="https://user-images.githubusercontent.com/35224544/210494605-ced00a43-f3fd-4a01-b4fd-d8ed483a8bfe.png">

**Reply by kumilingus** [reply]
I'm sorry, what does it have to do with links and markers? The `link` in JointJS terminology is a connection between 2 cells or points (usually 2 elements).

Anyway, here's an [example](https://codepen.io/jointjs/pen/vYaXJXr/5b9b9330a49e8ad1d13f164cdaca9735) of how to restrict the movement of embedded elements.

**Reply by slvamegestic** [reply]
Thank you for your response.I got the point. The reason I am stuck with the link is that I am only able to create a custom element using SVG via a link path and image. Having seen your example, is there any way to create an image similar to the one my above image? Please help me if it's possible

**Reply by kumilingus** [reply]
> The reason I am stuck with the link is that I am only able to create a custom element using SVG via a link path and image. 

I don't understand.

> Having seen your example, is there any way to create an image similar to the one my above image? 

Didn't I just create a similar example [here](https://codepen.io/jointjs/pen/vYaXJXr/5b9b9330a49e8ad1d13f164cdaca9735)?
Where is the link on the image?

**Reply by slvamegestic** [reply]
yes, you did. But my prob is cannot able to create bordered Rectable and some other modifications. I don't want to use HeaderedRectangle, or Rectangle in my code. I want to create an element by using "new joint.shapes.standard.Link()" like the code above and want to set the remove listener and bounding box with interactive. if is not possible to do that with link that's ok to be confirmed with you and one more thing please let me know how to set the remove link attr "new joint.shapes.standard.Link()". I am newbie to jointJs, so please forgive me if I asked an unnecessary question.

**Reply by kumilingus** [reply]
I would not recommend using links as containers. It sounds like a hack. A link is a relationship between 2 objects.

Even though it is possible to define the link labels to look the way you need (https://jsfiddle.net/kumilingus/dy3v8fr7/) to embed other elements inside the link label is not supported.

To be honest I don't understand what you mean by "set the remove listener and bounding box with interactive.".

Please try to describe the issue clearly, provide context, create a minimal reproducible example if applicable (more info on how to write good questions cab be found for instance [here](https://www.freecodecamp.org/news/how-to-ask-good-technical-questions/)).

If you have other questions (which are not related to this issue) please create a new discussion.

**Reply by slvamegestic** [reply]
@kumilingus 
I just noticed the message.This has been noted and I will take action as needed. Thank you very much for your time. Should I have any further thoughts, I will come up with another good topic for discussion.

For that remove listener, I mean that when cells are removed from the paper, the custom function should be executed in order to customize the paper. Anyway, I have done that using graph.on('remove', callback). Thanks
